src/System.Text.Json/src/System.Text.Json.csproj

```diff
@@ -22,7 +22,9 @@
     <Compile Include="System\Text\Json\JsonException.cs" />
     <Compile Include="System\Text\Json\JsonHelpers.cs" />
     <Compile Include="System\Text\Json\JsonHelpers.Date.cs" />
+    <Compile Include="System\Text\Json\Node\JsonNode.cs" />
```

Comment:
I just noticed you added these files under Node subfolder. Traditionally we have tried to have the folder structure mimic the namespace structure as well as close as possible, but I do noticed that this project in particular isn't really following the convention. @ahsonkhan I guess it's really up to you if you want to remove the Node subfolder or not. If we do decide to keep the folder, I'm not sure if Node is the right name for it.

Reply:
This has come up before and we mainly deferred it to avoid churn/refactoring as we focused on stabilizing for 3.0 (and to avoid unnecessary merge conflicts/loss of history): https://github.com/dotnet/corefx/issues/38408#issuecomment-500544013

We can certainly clean up the file structure to be consistent after 3.0 ships.

Based on what we discussed offline, we should also consider moving the new JsonNode related files being added in this PR to its own `ItemGroup` (to keep master -> release/3.0 ports and cherry-picks of other, unrelated changes easier and without unnecessary merge conflict resolution).